estimagic/visualization/comparison_plot.py

```diff
@@ -0,0 +1,440 @@
+    Build the DataFrame combining all DataFrames from the results dictionary.
+
+    Args:
+        res_dict (dict): see comparsion_plot docstring
+    """
+    df = pd.DataFrame()
+    for model, model_dict in res_dict.items():
+        result_df = _prep_result_df(model_dict, model)
+        df = df.append(result_df, sort=False)
+
+    if "group" not in df.columns:
+        df["group"] = "all"
+    if "conf_int_upper" not in df.columns:
+        df["conf_int_upper"] = np.nan
+    if "conf_int_lower" not in df.columns:
+        df["conf_int_lower"] = np.nan
+
+    return df.reset_index(drop=True)
+
+
+def _prep_result_df(model_dict, model):
+    result_df = model_dict["result_df"].reset_index()
+    result_df["model"] = model
+    name_cols = [x for x in ["group", "name"] if x in result_df.columns]
+    result_df["full_name"] = result_df[name_cols].apply(
+        lambda x: index_element_to_string(tuple(x)), axis=1
+    )
+    if "model_class" in model_dict.keys():
+        model_class = model_dict["model_class"]
+        result_df["model_class"] = model_class
+    else:
+        result_df["model_class"] = "no class"
+    return result_df
+
+
+def _create_bounds_and_rect_widths(df):
+    """Calculate lower and upper bounds and the width of the "histogram" bars."""
+    grouped = df.groupby("group")
+    upper = grouped[["conf_int_upper", "final_value"]].max().max(axis=1)
+    lower = grouped[["conf_int_lower", "final_value"]].min().min(axis=1)
+    rect_widths = 0.02 * (upper - lower)
+
+    return lower, upper, rect_widths
+
+
+def _determine_plot_heights(df, figure_height):
+    grouped = df.groupby("group")
+    figure_height = _determine_figure_height(df, figure_height)
+    nr_params = grouped["full_name"].unique().apply(len)
+    height_shares = nr_params / nr_params.sum()
+    return (height_shares * figure_height).astype(int)
+
+
+def _determine_figure_height(df, figure_height):
+    if figure_height is None:
+        nr_models = len(df["model"].unique())
+        nr_params = len(df["full_name"].unique())
+        figure_height = 8 * max(min(nr_models, 60), 10) * nr_params
+    return figure_height
+
+
+def _add_plot_specs_to_df(df, rect_widths, lower, upper, color_dict):
+    """Add color column, dodge column and binned_x colomun to *df*."""
+    _add_color_column(df, color_dict)
+    df["dodge"] = 0.5
+    for group in df["group"].unique():
+        rect_width = rect_widths.loc[group]
+        bins = np.arange(
+            start=lower.loc[group] - 2 * rect_width,
+            stop=upper.loc[group] + 2 * rect_width,
+            step=rect_width,
+        )
+        param_names = df[df["group"] == group]["full_name"].unique()
+        for param in param_names:
+            _add_dodge_and_binned_x(df, param, bins)
+    df["binned_x"].fillna(df["final_value"], inplace=True)
+
+
+def _add_color_column(df, color_dict):
+    if color_dict is None:
+        color_dict = {}
+    models = df["model_class"].unique()
+    for m in models:
```

Comment:
``color_dict = {color_dict.get(m, "#035096") for m in models}``

Reply:
Much nicer! Replaced in 604f9b37a2dc4bc5123ddc29a8fceba432d60334.